### DISCUSSION TITLE
Please share your coding approach to modelling!

### DISCUSSION THREAD

**Discussion by timothy-holmes** [pyswmm/pyswmm#526] [General]
> New opportunity awaits when you consider embracing a coding approach to your modeling practice

Riffing off this statement from @bemcdonnell in a [SWMM_USERS] post, I want to start a discussion about how pyswmm (and associated tools) is opening up new ways of doing things.

I haven't been able to replicate my modelling done InfoWorks ICM to pyswmm, but my eyes are open to the possibilities! Keen to hear about transformed workflow.

**Comment by bemcdonnell** [comment]
@timothy-holmes as we start working with larger and larger data sets, it becomes harder and takes longer to do simple things.  The whole power of inviting pyswmm into your workflow is the automation potential.  Furthermore, the ability to do optimization is very helpful

https://www.youtube.com/channel/UCv-OYsz2moiMRzZIRhqbpHA

Check out the "Why PySWMM" video on YouTube.  Also, there many training resources on our website with downloadable examples.  I'd be interested in hearing what your perspective is after diving into these resources!
